src/AgoraGovernor.sol

```diff
@@ -562,6 +564,13 @@ contract AgoraGovernor is
         bytes[] memory calldatas,
         bytes32 descriptionHash
     ) public onlyAdminOrTimelock returns (uint256) {
+        uint256 proposalId = hashProposal(targets, values, calldatas, descriptionHash);
+        address sender = _msgSender();
+        require(
+            sender == admin || sender == timelock() || sender == _proposals[proposalId].proposer,
+            "Governor: only admin, governor timelock, or proposer can cancel"
```

Comment:
Since we are starting to brush against the gas limit with the scopes work can this be a custom error instead? w/o a revert message?